crates/cheatnet/src/cheatcodes/deploy.rs

```diff
@@ -25,11 +23,11 @@ use starknet_api::core::{ClassHash, ContractAddress};
 use starknet_api::transaction::Calldata;
 
 use super::CheatcodeError;
-use crate::rpc::{CallContractFailure, ResourceReport};
+use crate::rpc::CallContractFailure;
 
 #[derive(Debug)]
 pub struct DeployCallPayload {
-    pub resource_report: ResourceReport,
+    pub used_resources: ExecutionResources,
```

Comment:
Do we ever want to ignore `deploy's` syscall cost? If so - do we have an issue for that? If not - we don't need this struct (we should have an issue for that too)
cc @Arcticae @MaksymilianDemitraszek